doc/source/reference/api/models/llama.rst

```diff
@@ -0,0 +1,238 @@
+.. _api-models-llama:
+
+=====================
+fairseq2.models.llama
+=====================
+
+.. currentmodule:: fairseq2.models.llama
+
+The LLaMA module provides support for LLaMA language models from Meta AI.
+It includes model configurations, hub access, tokenizers, and utilities for loading and working with LLaMA models.
+
+Quick Start
+-----------
+
+.. code-block:: python
+
+    from fairseq2.models.llama import get_llama_model_hub, get_llama_tokenizer_hub
+
+    # Get the model hub
+    hub = get_llama_model_hub()
+
+    # Load a model
+    model = hub.load_model("llama3_2_1b")
+
+    # Load corresponding tokenizer (uses HuggingFace tokenizer by default)
+    tokenizer = get_llama_tokenizer_hub().load_tokenizer("llama3_2_1b")
+
+    # Generate some text
+    text = "The future of AI is"
+    encoder = tokenizer.create_encoder()
+    encoded = encoder(text)
+    # ... model inference code ...
+
+Tokenizer
+---------
+
+LLaMA tokenizer in fairseq2 supports multiple implementations:
+
+1. **HuggingFace Tokenizer (Default)**:
+    The default and recommended implementation using HuggingFace's tokenizer.
+
+    Asset Card Example:
+
+    .. code-block:: yaml
+
+        name: llama3
+        tokenizer: "/path/to/Llama-3.1-8B"  # HuggingFace tokenizer directory
+        tokenizer_family: llama
+
+    Directory structure should contain e.g. ``config.json``, ``tokenizer.json``, ``tokenizer_config.json``, ``special_tokens_map.json``.
+
+2. **Tiktoken Implementation**:
+    Legacy implementation using Tiktoken.
```

Comment:
I would not say "legacy" since this still has the extra `split_regex` feature that is not available in the HG version.